edge-modules/TwinTester/Settings.cs

```diff
@@ -20,47 +20,56 @@ public class Settings
+
+                string analyzerUrl = configuration.GetValue<string>("AnalyzerUrl", "http://analyzer:15000");
+                string reporterUrl = configuration.GetValue<string>("testResultCoordinatorUrl", analyzerUrl);
 
                 return new Settings(
                     configuration.GetValue<string>("IOTEDGE_DEVICEID", string.Empty),
                     moduleId,
+                    targetModuleId,
                     configuration.GetValue<int>("TwinUpdateSize", 1),
                     configuration.GetValue<TimeSpan>("TwinUpdateFrequency", TimeSpan.FromSeconds(10)),
                     configuration.GetValue<TimeSpan>("TwinUpdateFailureThreshold", TimeSpan.FromMinutes(1)),
                     configuration.GetValue<TransportType>("TransportType", TransportType.Amqp_Tcp_Only),
-                    configuration.GetValue<Uri>("AnalyzerUrl", new Uri("http://analyzer:15000")),
+                    new Uri(reporterUrl),
                     configuration.GetValue<string>("ServiceClientConnectionString"),
                     configuration.GetValue<string>("StoragePath"),
                     configuration.GetValue<bool>("StorageOptimizeForPerformance", true),
-                    configuration.GetValue<TestMode>("TestMode", TestMode.TwinAllOperations));
+                    configuration.GetValue<TestMode>("TestMode", TestMode.TwinAllOperations),
+                    Option.Maybe(configuration.GetValue<string>("trackingId", null)));
             });
 
         Settings(
             string deviceId,
             string moduleId,
+            string targetModuleId,
             int twinUpdateSize,
             TimeSpan twinUpdateFrequency,
             TimeSpan twinUpdateFailureThreshold,
             TransportType transportType,
-            Uri analyzerUrl,
+            Uri reporterUrl,
             string serviceClientConnectionString,
             string storagePath,
             bool storageOptimizeForPerformance,
-            TestMode testMode)
+            TestMode testMode,
+            Option<string> trackingId)
         {
             this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
             this.ModuleId = Preconditions.CheckNonWhiteSpace(moduleId, nameof(moduleId));
+            this.TargetModuleId = Preconditions.CheckNonWhiteSpace(moduleId, nameof(targetModuleId));
             this.TwinUpdateSize = Preconditions.CheckRange(twinUpdateSize, 1);
             this.TwinUpdateFrequency = Preconditions.CheckNotNull(twinUpdateFrequency);
             this.TwinUpdateFailureThreshold = Preconditions.CheckNotNull(twinUpdateFailureThreshold);
             this.TransportType = Preconditions.CheckNotNull(transportType);
-            this.AnalyzerUrl = Preconditions.CheckNotNull(analyzerUrl, nameof(analyzerUrl));
+            this.ReporterUrl = Preconditions.CheckNotNull(reporterUrl, nameof(reporterUrl));
```

Comment:
```suggestion
            this.ReporterUrl = Preconditions.CheckNonWhiteSpace(reporterUrl, nameof(reporterUrl));
```

Reply:
reporterUrl has Uri type, for sure is not empty